restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +91,90 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  /**
+   *
+   * @param dataSchema
+   * @param dataMap
+   * @return a new different data map that contains original data plus the default values
+   */
+  public static DataMap fillInDefaultValues(DataSchema dataSchema, DataMap dataMap)
+  {
+    DataMap dataDefaultFilled = new DataMap();
+    if (dataMap != null)
+    {
+      dataDefaultFilled.putAll(dataMap);
+    }
+
+    if (dataSchema instanceof TyperefDataSchema)
+    {
+      TyperefDataSchema typerefDataSchema = (TyperefDataSchema) dataSchema;
+      return fillInDefaultValues(typerefDataSchema.getDereferencedDataSchema(), dataDefaultFilled);
+    }
+    else if (dataSchema instanceof RecordDataSchema)
+    {
+      RecordDataSchema recordDataSchema = (RecordDataSchema) dataSchema;
+      for (RecordDataSchema.Field field : recordDataSchema.getFields())
+      {
+        DataSchema fieldSchema = field.getType();
+        if (fieldSchema instanceof RecordDataSchema)
+        {
+          if (dataDefaultFilled.containsKey(field.getName()))
+          {
+            DataMap fieldDefault = (DataMap) field.getDefault();
+            DataMap fieldDataAssigned = (DataMap) dataDefaultFilled.get(field.getName());
+            if (fieldDataAssigned != null)
+            {
+              DataMap fieldFilled = fillInDefaultValues(fieldSchema, fieldDataAssigned);
+              dataDefaultFilled.put(field.getName(), fieldFilled);
```

Comment:
redundant as you are updating the field value inplace

Reply:
Just to be sure I get you correctly, I don't get why this is redundant, because as for now it is not `in place` yet.